plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -347,12 +347,50 @@ func NewSelectorMatchPredicate(info NodeInfo) algorithm.FitPredicate {
 	return selector.PodSelectorMatches
 }
 
+// New versions of the scheduler will apply their scheduling predicates
```

Comment:
This comment won't really make sense once this has been deployed for a while, and the full comment is in the design doc, so I would just replace it with "The pod can only schedule onto nodes that satisfy both hardNodeAffinity and nodeSelector."